### DISCUSSION TITLE
Setting abbreviations in ftplugin

### DISCUSSION THREAD

**Discussion by kejadlen** [rktjmp/hotpot.nvim#93] [Q&A] [answered]
So using the workaround from #41, I'm happy to have ftplugins in Fennel, but when I try setting abbreviations `(vim.cmd "iab <buffer> foo bar")` in the ftplugin, it doesn't take. Am I doing something wrong, is there some sort of interaction here that I'm tripping over, or is this expected?

**Comment by rktjmp** [comment]
Looks ok, put a print statement in there to check its running? It's definitely getting loaded? Is the buffer filetype correct to actually invoke the ftplugin?

**Reply by kejadlen** [reply]
Yup, I did do that, and could see the output in `:messages`, so I'm pretty sure it's running. FWIW, the end goal is to map `\lambda` to `λ` for Fennel, though once I have `iab` working in the `ftplugin`, I still need to overcome `\` not being usable with more than one character. But one thing at a time!

**Reply by rktjmp** [reply, marked as answer]
Sorry! Looks like I mislead you with that discussion, probably got my self tricked by a cached file perhaps, currently we only look for files in folders prefixed by `<rtp>/lua/` (or `fnl`), so when we look for `ftplugins.fennel` hotpot ends up looking for `lua|fnl/ftplugins/fennel.lua|fnl`.

I *can* make it look for files in "raw" rtp folders but I have to experiment as I think it would over match things if done without some care. I wish we could just register extra searcher with neovim itself. Probably I will have to explicitly whitelist some prefixes.

Anyway, if you shuffle your ftplugins file under `fnl/ftplugins/fennel.fnl` it should be found and run correctly. 

This should do your abbreviation, uses a global function for easy of use but you could probably stick it in a module and require it with `v:lua.require('xxx').x()` (maybe?).

```fennel
(fn _G.iab_lambda []
  (let [line (vim.fn.getline :.)
        col (vim.fn.col :.)
        ;; a b c \lambda
        ;;       ^ check here (may not exist)
        ;; but dont do anything if that under runs the first character
        offset (- col (length :lambda) 1)]
    (match [(< 0 offset) (string.sub line offset offset)]
      ;; replace term codes so the expr actually runs backspace, not inserts the string
      [true :\] (vim.api.nvim_replace_termcodes :<bs>λ true false true)
      [false _] :lambda)))
(vim.cmd "iab <buffer> <expr> lambda v:lua.iab_lambda()")
```

**Reply by kejadlen** [reply]
Oh, curses. The whole issue was me preferring to use the whole command in configuration files and not realizing that it's `iabbrev` and not `iabbreviation`. 🤦

Regardless, I had no idea how to use global functions from Fennel in Lua, so thank you very much for that snippet!

**Comment by rktjmp** [comment]
Discussion not relevant to v2.
